runtime/sema/type.go

```diff
@@ -4793,14 +4857,30 @@ func (t *ReferenceType) IndexingType() Type {
 	return referencedType.IndexingType()
 }
 
-func (*ReferenceType) Unify(_ Type, _ *TypeParameterTypeOrderedMap, _ func(err error), _ ast.Range) bool {
-	// TODO:
-	return false
+func (t *ReferenceType) Unify(
```

Comment:
Nice work implementing Unify/Resolve. Is this needed for the attachments feature?

Reply:
Doesn't actually seem like it; I thought it would be because of how I was planning on using references, but in actuality it doesn't seem necessary. If you think it doesn't belong in this PR I can revert the change here.

Reply:
No, it's great it got implemented, thank you! Please make sure we have tests for it as well, or maybe submit it as a separate PR, so this PR doesn't keep growing.